raiden/transfer/mediated_transfer/mediation_fee.py

```diff
@@ -37,6 +38,109 @@ def __call__(self, x: float) -> float:
         i = bisect_right(self.x_list, x) - 1
         return self.y_list[i] + self.slopes[i] * (x - self.x_list[i])
 
+    def __repr__(self) -> str:
+        return f"Interpolate({self.x_list}, {self.y_list})"
+
+
+def sign(x: float) -> int:
+    """ Sign of input, returns zero on zero input
+    """
+    if x == 0:
+        return 0
+    else:
+        return 1 if x > 0 else -1
+
+
+def _merge_x_values(
+    schedule_in: "FeeScheduleState",
+    schedule_out: "FeeScheduleState",
+    balance_in: Balance,
+    balance_out: Balance,
+    max_x: int,
+) -> List[float]:
+    """ Collect all relevant x values (edges of piece wise linear sections) """
+    assert schedule_in._penalty_func
+    assert schedule_out._penalty_func
+    all_x_vals = [x - balance_in for x in schedule_in._penalty_func.x_list] + [
+        balance_out - x for x in schedule_out._penalty_func.x_list
+    ]
+    limited_x_vals = (max(min(x, balance_out, max_x), 0) for x in all_x_vals)
+    return sorted(set(limited_x_vals))
+
+
+def _cap_fees(x_list: List[float], y_list: List[float]) -> Tuple[List[float], List[float]]:
+    """ Insert extra points for intersections with x-axis, see `test_fee_capping` """
+    x_list = copy(x_list)
+    y_list = copy(y_list)
+
+    for i in range(len(x_list) - 1):
+        y1, y2 = y_list[i : i + 2]
+        if sign(y1) * sign(y2) == -1:
+            x1, x2 = x_list[i : i + 2]
+            new_x = x1 + abs(y1) / abs(y2 - y1) * (x2 - x1)
+            new_index = bisect(x_list, new_x)
+            x_list.insert(new_index, new_x)
+            y_list.insert(new_index, 0)
+
+    # Cap points that are below zero
+    y_list = [max(y, 0) for y in y_list]
+    return x_list, y_list
+
+
+def _mediation_fee_func(
+    schedule_in: "FeeScheduleState",
+    schedule_out: "FeeScheduleState",
+    balance_in: Balance,
+    balance_out: Balance,
+    receivable: TokenAmount,
+    amount_with_fees: Optional[PaymentWithFeeAmount],
+    amount_without_fees: Optional[PaymentWithFeeAmount],
+    cap_fees: bool,
+) -> Interpolate:
+    """ Returns a function which calculates total_mediation_fee(x)
+
+    Either `amount_with_fees` or `amount_without_fees` must be given while the
+    other one is None. The returned function will depend on the value that is
+    not given.
+    """
+    assert amount_with_fees is None or amount_without_fees is None
```

Comment:
Can you put a condensed version of the last paragraph of the docstring as the assert's failure message?